.github/workflows/ci-pipeline.yml

```diff
@@ -59,4 +59,43 @@ jobs:
       - name: Run Snyk to check for vulnerabilities
         uses: snyk/actions/python-3.8@master
         env:
-          SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
\ No newline at end of file
+          SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
+
+  docker:
+    if: github.event_name == 'push' && github.ref_name == 'main'
+    needs: build
+    name: Build and publish to Docker
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Docker Login
+        uses: docker/login-action@v2.0.0
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_ACCESSTOKEN }}
+      
+      - name: Build and push Docker image
+        run: |
+          docker build --target production --tag ${{ secrets.DOCKERHUB_USERNAME }}/todo-app:latest .
+          docker push ${{ secrets.DOCKERHUB_USERNAME }}/todo-app:latest
+
+  heroku:
+    needs: docker
+    name: Publish to Heroku
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2         
+
+      - name: Pull and tag Docker image
+        run: |
+          HEROKU_API_KEY=${{ secrets.HEROKU_API_KEY }} heroku container:login
+          docker pull ${{ secrets.DOCKERHUB_USERNAME }}/todo-app:latest
+          docker tag ${{ secrets.DOCKERHUB_USERNAME }}/todo-app:latest registry.heroku.com/swish-todoapp/web
```

Comment:
Just a comment to say I like this pattern; keeping your publishing the image & releasing to Heroku as separate jobs, but taking advantage of pulling the image which is quite a bit quicker than building it. Nice 😄